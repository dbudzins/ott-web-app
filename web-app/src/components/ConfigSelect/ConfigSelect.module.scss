.dropdown {
  position: fixed;
  right: 0;
  bottom: 0;

  width: auto;

  margin-right: 8px;
  margin-left: auto;

  > div {
    min-width: 150px;
  }
}