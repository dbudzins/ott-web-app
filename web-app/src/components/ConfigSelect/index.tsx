import Dropdown from '../Dropdown/Dropdown';

import styles from './ConfigSelect.module.scss';

import { getStoredConfig, setStoredConfig } from '#src/utils/configOverride';
import { logDev } from '#src/utils/common';

const configs = import.meta.glob('../../../test-e2e/data/*.json');

const configOptions = Object.entries(configs).reduce<string[]>((acc, [pth]) => {
  let configName = '';

  try {
    const parts = pth.split('/');
    const fileName = parts.pop();
    configName = fileName?.split('.')[1] as string;
  } catch (err: unknown) {
    logDev('Config file could not be processed: ' + err);
  }

  return configName ? [...acc, configName] : acc;
}, []);

const ConfigSelect = () => {
  const selectedConfig = getStoredConfig() || '';

  const onChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    setStoredConfig(event.target.value);
    window.location.reload();
  };

  return (
    <Dropdown
      className={styles.dropdown}
      size="small"
      options={configOptions}
      defaultLabel={'Default config'}
      name="config-select"
      value={selectedConfig}
      onChange={onChange}
    />
  );
};

export default ConfigSelect;
