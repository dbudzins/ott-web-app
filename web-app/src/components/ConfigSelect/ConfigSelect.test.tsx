import React from 'react';

import ConfigSelect from './';

import { renderWithRouter } from '#test/testUtils';

describe('<ConfigSelect>', () => {
  test('renders and matches snapshot', () => {
    const { container } = renderWithRouter(<ConfigSelect />);

    expect(container).toMatchSnapshot();
  });
});
