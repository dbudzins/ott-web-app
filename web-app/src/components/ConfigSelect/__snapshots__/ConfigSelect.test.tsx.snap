// Vitest Snapshot v1

exports[`<ConfigSelect> > renders and matches snapshot 1`] = `
<div>
  <div
    class="container dropdown"
  >
    <div
      class="dropdown"
    >
      <select
        aria-required="false"
        class="select"
        id="1235"
        name="config-select"
      >
        <option
          class="option"
          value=""
        >
          Default config
        </option>
        <option
          class="option"
          value="test--accounts"
        >
          test--accounts
        </option>
        <option
          class="option"
          value="test--blender"
        >
          test--blender
        </option>
        <option
          class="option"
          value="test--dev"
        >
          test--dev
        </option>
        <option
          class="option"
          value="test--no-cleeng"
        >
          test--no-cleeng
        </option>
        <option
          class="option"
          value="test--no-watchlists"
        >
          test--no-watchlists
        </option>
        <option
          class="option"
          value="test--subscription"
        >
          test--subscription
        </option>
      </select>
    </div>
  </div>
</div>
`;
